security/cores.py

```diff
@@ -38,14 +31,59 @@ def display_as_code(value):
     return format_html('<code style="white-space:pre-wrap;">{}</code>', value) if value else value
 
 
-
 class DisplayRelatedObjectsMixin:
 
     @short_description(_('related objects'))
     def display_related_objects(self, obj, request):
         return render_model_objects_with_link(
             request,
-            [related_object.object for related_object in obj.related_objects.all() if related_object.object]
+            [related_object.object for related_object in obj.related_objects.exclude(
+                object_ct__in={
+                    ContentType.objects.get_for_model(model_class)
+                    for model_class in (CommandLog, InputLoggedRequest, OutputLoggedRequest,
+                                        CeleryTaskLog, CeleryTaskRunLog)
+                }
+            ) if related_object.object]
+        )
+
+    @short_description(_('source'))
+    def display_source(self, obj, request):
+        return render_model_objects_with_link(
+            request,
+            [related_object.object for related_object in obj.related_objects.filter(
+                object_ct__in={
+                    ContentType.objects.get_for_model(model_class)
+                    for model_class in (CommandLog, InputLoggedRequest, OutputLoggedRequest,
+                                        CeleryTaskLog, CeleryTaskRunLog)
+                }
+            ) if related_object.object]
+        )
+
+    @short_description(_('raised output logged requests'))
+    def display_output_logged_requests(self, obj, request):
+        return render_model_objects_with_link(
+            request,
+            [related_object for related_object in OutputLoggedRequest.objects.filter(
+                related_objects__object_id=obj.pk,
+                related_objects__object_ct=ContentType.objects.get_for_model(obj))]
+        )
+
+    @short_description(_('raised command logs'))
+    def display_command_logs(self, obj, request):
+        return render_model_objects_with_link(
+            request,
+            [related_object for related_object in CommandLog.objects.filter(
+                related_objects__object_id=obj.pk,
+                related_objects__object_ct=ContentType.objects.get_for_model(obj))]
+        )
+
+    @short_description(_('raised celery task logs'))
+    def display_celery_task_logs(self, obj, request):
+        return render_model_objects_with_link(
+            request,
+            [related_object for related_object in CeleryTaskLog.objects.filter(
+                related_objects__object_id=obj.pk,
+                related_objects__object_ct=ContentType.objects.get_for_model(obj))]
```

Comment:
All of these three properties could share common logic through a helper function.